### PR TITLE
settings voor sendmail

### PR DESCRIPTION
**Pull request by NiekD**

### CODE REVIEW

**Comment by CurlyMoo** [comment]
Please enclose this in the EVENTS define in case users don't want to compile events support.

File: libs/config/settings.c [hunk 1]
```diff
@@ -435,6 +435,88 @@ static int settings_parse(JsonNode *root) {
...
+				logprintf(LOG_ERR, "config setting \"%s\" must contain an e-mail address", jsettings->key);
+				have_error = 1;
+				goto clear;
+			} else if(!jsettings->string_) {
+				logprintf(LOG_ERR, "config setting \"%s\" must contain an e-mail address", jsettings->key);
+				have_error = 1;
+				goto clear;
+			} else if(strlen(jsettings->string_) > 0) {
+				char validate[] = "^[a-zA-Z0-9_.]+@([a-zA-Z0-9]+.)+[a-zA-Z0-9]{2,3}$";
+				reti = regcomp(&regex, validate, REG_EXTENDED);
```
Comment (CurlyMoo) [inline libs/config/settings.c]: Regex don't work on FreeBSD and Windows, so please enclose.

File: libs/config/settings.c [hunk 2]
```diff
@@ -435,6 +435,88 @@ static int settings_parse(JsonNode *root) {
...
+					settings_add_string(jsettings->key, jsettings->string_);
+				}
+			}
+		} else if(strcmp(jsettings->key, "mail-subject") == 0 ||
+					strcmp(jsettings->key, "smtp-password") == 0) {
+			if(jsettings->tag != JSON_STRING) {
+				logprintf(LOG_ERR, "config setting \"%s\" must contain a string", jsettings->key);
+				have_error = 1;
+				goto clear;
+			} else if(!jsettings->string_) {
```
Comment (CurlyMoo) [inline libs/config/settings.c]: Please use `== NULL` here (and add the specific value we are comparing to everywhere.

File: libs/config/settings.c [hunk 3]
```diff
@@ -533,4 +615,4 @@ void settings_init(void) {
 	config_settings->parse=&settings_parse;
 	config_settings->sync=&settings_sync;
 	config_settings->gc=&settings_gc;
-}
+}
```
Comment (CurlyMoo) [inline libs/config/settings.c]: Add a newline here.

**Comment by NiekD** [comment]
OK

CurlyMoo schreef op 17-3-2015 om 19:09:

File: libs/config/settings.c (hunk 1 above)
Comment (NiekD) [inline libs/config/settings.c]: OK

CurlyMoo schreef op 17-3-2015 om 19:09:

File: libs/config/settings.c (hunk 2 above)
Comment (NiekD) [inline libs/config/settings.c]: Sorry, I don't understand this one


CurlyMoo schreef op 17-3-2015 om 19:10:

File: libs/config/settings.c (hunk 3 above)
Comment (NiekD) [inline libs/config/settings.c]: OK

CurlyMoo schreef op 17-3-2015 om 19:11:

File: libs/config/settings.c (hunk 2 above)
Comment (CurlyMoo) [inline libs/config/settings.c]: I've been changing this slowly everywhere. The problem with `if(!jsettings->string_)` is that it doesn't tell us to what we are comparing. Is it a pointer, decimals, etc? So by making it `jsettings->string_ == NULL` we know we are working with a char pointer here.

Comment (NiekD) [inline libs/config/settings.c]: OK, clear now

CurlyMoo schreef op 17-3-2015 om 21:22:

**Pull request closed without merging**